rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/TechnologyIdentified.java

```diff
@@ -0,0 +1,146 @@
+package org.jboss.windup.rules.apps.javaee;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.operation.iteration.AbstractIterationOperation;
+import org.jboss.windup.graph.model.FileLocationModel;
+import org.jboss.windup.graph.model.ProjectModel;
+import org.jboss.windup.graph.model.WindupVertexFrame;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.reporting.model.TagSetModel;
+import org.jboss.windup.reporting.service.TagSetService;
+import org.jboss.windup.rules.apps.javaee.model.stats.TechnologyUsageStatisticsModel;
+import org.jboss.windup.util.Logging;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+/**
+ * Adds the specified statistics and tag information regarding an technology that has been located in the analyzed application.
+ *
+ * If no count is specified, then a default of "1" is assumed.
+ *
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public class TechnologyIdentified extends AbstractIterationOperation<WindupVertexFrame>
+            implements TechnologyIdentifiedWithCount, TechnologyIdentifiedWithName
+{
+    public static final int DEFAULT_COUNT = 1;
+
+    private static Logger LOG = Logging.get(TechnologyIdentified.class);
+
+    private String technologyName;
+    private Set<String> tags = new HashSet<>();
+    private int count = DEFAULT_COUNT;
+
+    /**
+     * Creates the operation with the given technology name.
+     */
+    public TechnologyIdentified(String technologyName)
+    {
+        this.technologyName = technologyName;
+    }
+
+    /**
+     * Creates the operation with the given input variable and technology name.
+     */
+    public TechnologyIdentified(String variableName, String technologyName)
+    {
+        super(variableName);
+        this.technologyName = technologyName;
+    }
+
+    /**
+     * Create the operation with the given technology name.
+     */
+    public static TechnologyIdentifiedWithName named(String technologyName)
+    {
+        return new TechnologyIdentified(technologyName);
+    }
+
+    /**
+     * Sets the number of items that have been found by this operation.
+     *
+     * This is optional, and if left unspecified, it will default to {@link TechnologyIdentified#DEFAULT_COUNT}.
+     */
+    public TechnologyIdentifiedWithCount numberFound(int count)
+    {
+        this.count = count;
+        return this;
+    }
+
+    /**
+     * Specifies a tag to associate with this technology.
+     */
+    public TechnologyIdentified withTag(String tag)
+    {
+        this.tags.add(tag);
+        return this;
```

Comment:
Maybe rename this method to `addTag`? It is not clear `withTag` adds tag to collection.

Reply:
It is just to have a fluent API. See above the comment regarding fluent APIs.